src/pages/index.js

```diff
@@ -62,31 +101,99 @@ function displayImgModal(e) {
   imgFormModal.open();
 }
 
+function displayAvatarModal() {
+  cardFormValidator.toggleButtonState();
+  avatarFormModal.open();
+}
 // function that closes img display modal
 function handleCardClick(imageUrl, caption) {
   imgDisplayPopup.open(imageUrl, caption);
 }
 
 //function that saves profile modal inputs
 
-function handleProfileFormSubmit(formData) {
-  const { name, description } = formData;
-  userInfo.setUserInfo(name, description);
-  profileFormModal.close();
+function handleProfileFormSubmit({ name, description }) {
+  profileFormModal.setLoading(true);
+  api
+    .updateUserInfo(name, description)
+    .then(() => {
+      userInfo.setUserInfo(name, description);
+      profileFormModal.close();
+    })
+    .catch((error) => {
+      console.log(error);
+    })
+    .finally(() => {
+      profileFormModal.setLoading(false);
+    });
 }
 
 //function that save imgs modal inputs
 
-function handleImgFormSubmit(inputValues) {
-  const { name, link } = inputValues;
-  const newCard = createCard({ name, link });
-  section.addItem(newCard);
-  imgFormModal.close();
+function handleImgFormSubmit({ name, link }) {
+  api
+    .createCards({ name, link })
+    .then((cardData) => {
+      const newCard = createCard(cardData);
+      section.addItem(newCard);
+      imgFormModal.close();
+    })
+    .catch(console.error);
+}
+
+//funtion to change avatar
+
+function handleAvatarFormSubmit(inputValue) {
+  avatarFormModal.setLoading(true);
+  api
+    .updateUserAvatar(inputValue.link)
+    .then(() => {
+      userInfo.setUserAvatar(inputValue.link);
+      avatarFormModal.close();
+    })
+    .catch((error) => {
+      console.log(error);
+    })
+    .finally(() => {
+      avatarFormModal.setLoading(false);
+    });
 }
 
 //function that creates cards
 function createCard(item) {
-  const card = new Card(item, "#profile__card-template", handleCardClick);
+  const card = new Card(
+    item,
+    "#profile__card-template",
+    handleCardClick,
+    function handleDeleteCard() {
+      deleteCardPopup.setConfirmAction(() => {
+        deleteCardPopup.setLoading(true);
+
+        //Executing the API Call within the confirmation action
+        api
+          .deleteCard(item._id)
+          .then((res) => {
+            card.deleteCardElement(res._id);
+          })
+          .catch((error) => {
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors